bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductsRetriever.java

```diff
@@ -37,11 +71,16 @@ protected ProductInterfaceQueryDefinition generateProductQuery() {
                 .urlRewrites(uq -> uq.url())
                 .priceRange(r -> r
                     .minimumPrice(generatePriceQuery()))
-                .onConfigurableProduct(cp -> cp
-                    .variants(v -> v
-                        .product(generateSimpleProductQuery()))
-                    .priceRange(r -> r
-                        .maximumPrice(generatePriceQuery())));
+                .onConfigurableProduct(cp -> {
+                    if (!isCategoryQuery()) {
+                        cp.variants(v -> v.product(generateSimpleProductQuery()));
+                    }
+                    cp.priceRange(r -> r.maximumPrice(generatePriceQuery()));
+                });
+
+            if (isCategoryQuery()) {
```

Comment:
This queries the categories for each product and so will add some overhead/redundancy to the response.

Alternatively you could query the category individually (still in the same query) like so:

```{  
 	products(filter:{category_uid:{eq:"Mzk="}}) { ... }
  categories(filters:{category_uid:{eq:"Mzk="}}) {
    items {
      uid
      url_key
      url_path
    }
  }
}```

Reply:
Not sure if that's a real improvement.

Reply:
Products can be in many categories and currently this is requesting all of them. This overhead on the wire but also  the unnecessary logic to pick the selected category from the list of categories retrieved from the product can be reduced to retrieve only a single category (the selected one).

Reply:
AFAIK, multiple queries in one request don't play well with graphql caching.
Hitting the built-in max 50 products limit would likely happen in case of an attack on the system. Most often you are dealing only with a few products here.

Reply:
we should discuss this again.